### PR TITLE
Add nested type description section

### PR DESCRIPTION
**Pull request by methylDragon**
Ditto the title

### CODE REVIEW

**Comment by methylDragon** [comment]
Some things to consider (I added them in the REP also):
- I think we should have `strings` for `field_types` to support custom type names (space-saving alternative on slack)
- I think we should have `type_name` to store the name of the type (so we can use it to key the `referenced_type_descriptions` array)

File: rep-2011.rst [hunk 1]
```diff
@@ -290,17 +290,107 @@ And the ``IndividualTypeDescription`` type:
...
+
+.. code::
+
+  # A: TypeDescription
+  type_description: A_IndividualTypeDescription
+  referenced_type_descriptions: [B_IndividualTypeDescription,
+                                 C_IndividualTypeDescription,
+                                 D_IndividualTypeDescription]
+
+With the referenced type descriptions accessible as ``IndividualTypeDescription`` types in the ``referenced_type_descriptions`` field of ``A`` (and also for `A`, its own ``type_description`` field.
```
Comment (wjwwood) [inline rep-2011.rst]: unmatched `(`

**Review by wjwwood — APPROVED** [review]

File: rep-2011.rst (hunk 1 above)
Comment (methylDragon) [reply rep-2011.rst]: https://github.com/wjwwood/rep/pull/4/commits/6c1e27132dbf2c2b0be24c4c5b9bed266056d8db

**Comment by methylDragon** [comment]
See notes in introspection API section.

It's changed a lot to support nested types, and I'm unsure if I'm making good decisions...

**Pull request merged**